_overviews/scala3-macros/tutorial/reflection.md

```diff
@@ -75,6 +98,46 @@ def g[T: Type](using Quotes) =
   ...
 ```
 
+## Symbols
+
+The APIs of `Term` and `TypeTree` are relatively *closed* in the sense that methods produce and accept values
+whose types are defined in the API. You might notice however the presence of `Symbol`s which identify definitions.
+
+Both `Term` or `TypeRepr` (therefore `Expr` and `Type`) have an associated symbol.
+`Symbol`s make it possible to compare two definitions using `==` to know if they are the same.
+In addition `Symbol` exposes and is used by many useful methods. For example:
+
+ - `declaredFields` and `declaredMethods` allow you to iterate on the fields and members defined inside a symbol
+ - `flags` allows you to check multiple properties of a symbol
+ - `companionObject` and `companionModule` provide a way to jump to and from the companion object/class.
+ - `TypeRepr.baseClasses` returns the list of symbols of classes extended by a type. 
+ - `Symbol.pos` gives you access to the position where the symbol is defined, the source code of the definition
+ and even the filename where the symbol is defined.
+ - and many useful others that you can find in [`SymbolMethods`](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$SymbolMethods.html)
+
+### To Symbol and back
+
+ - `TypeRepr.typeSymbol` returns the symbol of the type represented by `TypeRepr`. The recommended way to obtain a `Symbol` given a `Type[T]` is `TypeRepr.of[T].typeSymbol`
+ - For a singleton type, `TypeRepr.termSymbol` returns the symbol of the underlying object or value.
+ - `TypeRepr.memberType(symbol)` returns the `TypeRepr` of the provided symbol
+ - `Tree.symbol` returns the symbol associated to a tree. Given that `Term <: Tree`,
+ `Expr.asTerm.symbol` is the best way to obtain the symbol associated to an `Expr[T]`
+ - `Symbol.tree` returns the `Tree` associated to the symbol. Be careful when using this
+ method as the tree for a symbol might not be defined. When the code associated to the symbol
+ is defined in a different moment than this access, if the `Yretain-trees` compilation option
+ is not used, then the `tree` of the symbol will not be available. Symbols originated from
+ Java code do not have an associated `tree`.
+
+## Suggestion and anti-patterns
```

Comment:
Those should be moved to https://docs.scala-lang.org/scala3/guides/macros/best-practices.html. There we can expand a bit each point and add some examples if possible.

We could leave a link to that page here.

Reply:
I moved a few points to best-practices.md and provided small examples